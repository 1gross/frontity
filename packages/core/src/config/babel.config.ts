import { TransformOptions } from "babel-core";
import { Env, Mode, BabelConfigs } from "./types";

const targets = {
  module: { esmodules: true },
  es5: {
    browsers: [
      "and_chr >= 67",
      "and_ff >= 18",
      "and_uc >= 11.8",
      "android >= 67",
      "not android <= 4.4.4",
      "chrome >= 49",
      "edge >= 12",
      "firefox >= 18",
      "ios_saf >= 10",
      "not op_mini all",
      "op_mob >= 46",
      "opera >= 36",
      "safari >= 10",
      "samsung >= 5"
    ]
  },
  node: { node: "8.10" }
};

export default (mode: Mode): BabelConfigs => {
  const getConfig = (env: Env): TransformOptions => {
    const presets = [
      "@babel/preset-typescript",
      ["@babel/preset-env", { useBuiltIns: "usage", targets: targets[env] }],
      "@babel/preset-react"
    ];
    const plugins = [
      "@babel/plugin-proposal-object-rest-spread",
      "@babel/plugin-proposal-class-properties"
    ];
    if (mode === "development" && env === "module")
      plugins.push("react-hot-loader/babel");
    return {
      presets,
      plugins
    };
  };

  return {
    module: getConfig("module"),
    es5: getConfig("es5"),
    node: getConfig("node")
  };
};
