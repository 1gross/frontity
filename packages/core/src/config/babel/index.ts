import { TransformOptions } from "babel-core";
import { Target, Mode, BabelConfigs } from "../../types";

const targets = {
  // Browsers with <script type="module"></script> support.
  module: { esmodules: true },
  // Browsers with Proxy support, which is needed by Overmind.
  // For older browsers (ie8-11) we support AMP fallback.
  es5: {
    browsers: [
      "and_chr >= 67",
      "and_ff >= 18",
      "and_uc >= 11.8",
      "android >= 67",
      "not android <= 4.4.4",
      "chrome >= 49",
      "edge >= 12",
      "firefox >= 18",
      "ios_saf >= 10",
      "not op_mini all",
      "op_mob >= 46",
      "opera >= 36",
      "safari >= 10",
      "samsung >= 5"
    ]
  },
  // Node version used by AWS Lambda.
  node: { node: "8.10" }
};

export default ({ mode }: { mode: Mode }): BabelConfigs => {
  const getConfig = (target: Target): TransformOptions => {
    const presets = [
      // Instead of using a TS transpiler, this removes the typescript code.
      "@babel/preset-typescript",
      [
        "@babel/preset-env",
        {
          targets: targets[target]
          // esModules and CJS don't work well together so this is here
          // to be able to parse node_modules, where there are many modules
          // still using CJS. It will probably break tree shaking, so we need
          // to check that later. Another option would be the opposite:
          // https://www.npmjs.com/package/babel-plugin-transform-commonjs-es2015-modules
          // modules: target !== "node" ? "false" : "commonjs"
        }
      ],
      "@babel/preset-react"
    ];
    const plugins = [
      // Support for dynamic imports: import("./my-file")
      "@babel/plugin-syntax-dynamic-import",
      // Needed for loadable-component SSR.
      "@loadable/babel-plugin",
      // Support for the rest spread: { ...obj }
      "@babel/plugin-proposal-object-rest-spread",
      // Support for the class props: class MyClass { myProp = 'hi there' }
      "@babel/plugin-proposal-class-properties"
    ];
    return {
      presets,
      plugins
    };
  };

  return {
    module: getConfig("module"),
    es5: getConfig("es5"),
    node: getConfig("node")
  };
};
