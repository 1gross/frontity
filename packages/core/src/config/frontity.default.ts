import FrontityConfig from '.';

const defaultConfig: FrontityConfig = {
  headers: {
    'Access-Control-Allow-Origin': '*',
    'Origin, X-Requested-With, Content-Type': 'Accept',
  },
  staticHeaders: {},
  dynamicHeaders: {},
  robots: 'User-agent: *\nDisallow:',
};

export default defaultConfig;
