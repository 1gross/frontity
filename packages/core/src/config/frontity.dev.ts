import FrontityConfig from '.';

const config = (config: FrontityConfig) => {
  config.staticHeaders['Cache-Control'] = 'no-cache, no-store, must-revalidate';
  config.dynamicHeaders['Cache-Control'] =
    'no-cache, no-store, must-revalidate';
};

export default config;
