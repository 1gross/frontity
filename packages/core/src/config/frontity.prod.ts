import FrontityConfig from '.';

const config = (config: FrontityConfig) => {
  config.staticHeaders['Cache-Control'] =
    'public, max-age=31536000, s-maxage=31536000, immutable';
  config.dynamicHeaders['Cache-Control'] =
    'public, max-age=0, s-maxage=120, stale-while-revalidate=31536000, stale-if-error=31536000';
};

export default config;
