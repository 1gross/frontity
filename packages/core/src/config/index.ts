import { Mode, FrontityConfig } from "./types";
import getHeaders from "./headers";
import getBabel from "./babel";
import getWebpack from "./webpack";
import getRobots from "./robots";

export default ({ mode }: { mode: Mode }): FrontityConfig => {
  const headers = getHeaders({ mode });
  const babel = getBabel({ mode });
  const webpack = getWebpack({ mode, babel });
  const robots = getRobots({ mode });
  return {
    headers,
    babel,
    webpack,
    robots
  };
};
