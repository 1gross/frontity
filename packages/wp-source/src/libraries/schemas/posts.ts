import { schema } from "normalizr";
import { taxonomies } from "./taxonomies";
import { author } from "./authors";
import { attachment } from "./attachments";
import { normalize } from "../route-utils";

export const postType = new schema.Entity("postType");

export const post = new schema.Entity(
  "post",
  {},
  {
    processStrategy(entity) {
      const result = { ...entity };
      result.link = normalize(result.link);
      return result;
    }
  }
);

post.define({
  _embedded: {
    author: [author],
    type: [postType],
    "wp:featuredmedia": [attachment],
    "wp:contentmedia": [[attachment]],
    "wp:term": taxonomies
  }
});

export const posts = new schema.Array(post);
