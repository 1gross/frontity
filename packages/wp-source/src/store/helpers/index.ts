export const getIdBySlug = (
  entityMap: {
    [id: number]: {
      slug: string;
      id: number;
    };
  },
  slug: string
): number | undefined => {
  const entity = Object.values(entityMap).find(e => e.slug === slug);
  return entity && entity.id;
};

export const getTotal = (response: Response): number =>
  parseInt(response.headers.get("X-WP-Total"));

export const getTotalPages = (response: Response): number =>
  parseInt(response.headers.get("X-WP-TotalPages"));

export const normalizePath = (pathOrLink: string): string => {
  try {
    return new URL(pathOrLink).pathname;
  } catch (e) {
    // is not a URL
    return pathOrLink;
  }
};
