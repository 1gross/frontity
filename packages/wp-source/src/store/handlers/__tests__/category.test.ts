import { createOvermindMock } from "overmind";
import { namespaced } from "overmind/config";
import { state, actions, effects } from "../../";
import handler from "../category";

import { mockResponse, expectEntities } from "./mocks/helpers";
import posts from "../../__tests__/mocks/postsFromCategory.json";


describe("category", () => {
  test("doesn't exist in source.category", async () => {
    const path = "/category/nature/";
    const slug = "nature";

    // Mock resolver
    const resolver = {
      match: jest.fn().mockReturnValue({ handler, params: { slug } })
    };
    // Mock get responses
    const api = {
      getIdBySlug: jest.fn().mockResolvedValue(7),
      get: jest.fn().mockResolvedValue(
        mockResponse(posts, {
          "X-WP-Total": 10,
          "X-WP-TotalPages": 1
        })
      )
    };
    const config = namespaced({ source: { actions, state, effects } });
    const store = createOvermindMock(config, {
      source: { resolver, api, populate: effects.populate }
    });

    await store.actions.source.fetch(path);
    expect(api.get.mock.calls).toMatchSnapshot();
    expect(store.state.source.dataMap).toMatchSnapshot();

    // Check that all entities are being populated
    expectEntities(store.state.source);
  });

  test("exists in source.category but not in source.data", () => {});
  test("exists in source.data but doesn't have page", () => {});
  test("throws an error if it doesn't exist", () => {});
});
