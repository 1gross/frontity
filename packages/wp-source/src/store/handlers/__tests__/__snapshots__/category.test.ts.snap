// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`category doesn't exist in source.category 1`] = `
Array [
  Array [
    Object {
      "endpoint": "posts",
      "params": Object {
        "_embed": true,
        "categories": 7,
        "page": 1,
        "search": undefined,
      },
    },
  ],
]
`;

exports[`category doesn't exist in source.category 2`] = `
Object {
  "/category/nature/": Object {
    "id": 7,
    "isArchive": true,
    "isCategory": true,
    "isFetching": false,
    "isReady": true,
    "isTaxonomy": true,
    "pages": Array [
      Array [
        Object {
          "id": 57,
          "link": "https://test.frontity.io/2016/shinjuku-gyoen-national-garden/",
          "type": "post",
        },
        Object {
          "id": 54,
          "link": "https://test.frontity.io/2016/lower-antelope-canyon/",
          "type": "post",
        },
        Object {
          "id": 42,
          "link": "https://test.frontity.io/2016/kebumen/",
          "type": "post",
        },
        Object {
          "id": 36,
          "link": "https://test.frontity.io/2016/banff-national-park/",
          "type": "post",
        },
        Object {
          "id": 33,
          "link": "https://test.frontity.io/2016/the-al-hajar-mountains/",
          "type": "post",
        },
        Object {
          "id": 30,
          "link": "https://test.frontity.io/2016/machu-picchu/",
          "type": "post",
        },
        Object {
          "id": 26,
          "link": "https://test.frontity.io/2016/discovering-iceland/",
          "type": "post",
        },
        Object {
          "id": 23,
          "link": "https://test.frontity.io/2016/switzerlands-mountains/",
          "type": "post",
        },
        Object {
          "id": 12,
          "link": "https://test.frontity.io/2016/somewhere-in-romania/",
          "type": "post",
        },
        Object {
          "id": 1,
          "link": "https://test.frontity.io/2016/iceland/",
          "type": "post",
        },
      ],
    ],
    "taxonomy": "category",
    "total": 10,
    "totalPages": 1,
  },
}
`;

exports[`category doesn't exist in source.category 3`] = `
Object {
  "attachment": Array [
    6,
    13,
    24,
    27,
    34,
    37,
    45,
    55,
    77,
    80,
    91,
    93,
    95,
    103,
    107,
    108,
    109,
    111,
    113,
    116,
    118,
    126,
    132,
    133,
    134,
    185,
    193,
  ],
  "author": Array [
    1,
    2,
    3,
    4,
  ],
  "category": Array [
    3,
    4,
    5,
    6,
    7,
    8,
    56,
    57,
    58,
    59,
  ],
  "post": Array [
    1,
    12,
    23,
    26,
    30,
    33,
    36,
    42,
    54,
    57,
  ],
  "tag": Array [
    9,
    13,
    14,
    15,
    16,
    18,
    19,
    23,
    28,
    29,
    30,
    31,
    32,
    33,
    34,
    35,
    36,
    37,
    38,
    39,
    40,
    41,
    42,
    43,
    47,
    48,
    49,
    50,
  ],
}
`;
